packages/api/src/ontology/QueryDefinition.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface QueryDefinition<Q extends string> {
+  apiName: Q;
+  description?: string;
+  displayName?: string;
+  rid: string;
+  version: string;
+  parameters: Record<string, QueryParameterDefinition>;
+  output: QueryDataTypeDefinition;
+}
+
+export interface QueryParameterDefinition {
+  description?: string;
+  dataType: QueryDataTypeDefinition;
+}
+
+export interface QueryDataTypeDefinition {
+  type: QueryDataType;
+  multiplicity?: boolean;
+  nullable?: boolean;
+}
+
+export type QueryDataType =
+  | keyof ValidBaseQueryDataTypes
+  | ObjectQueryDataType<any>
+  | ObjectSetQueryDataType<any>
+  | SetQueryDataType
+  | UnionQueryDataType
+  | StructQueryDataType;
+
+export interface ValidBaseQueryDataTypes {
+  double: number;
+  float: number;
+  integer: number;
+  long: number;
+  boolean: boolean;
+  string: string;
+  date: Date;
+  timestamp: Date;
+  attachment: any; // TODO surely we can be more strict here
```

Comment:
I can add this to the gateway types but there is also a fun `TwoDimensionalAggregation` and `ThreeDimensionalAggregation` type we'll need to handle